driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -51,28 +50,48 @@
      */
     <R extends Expression> ArrayExpression<R> map(Function<? super T, ? extends R> in);
 
+    IntegerExpression size();
+
+    BooleanExpression any(Function<T, BooleanExpression> map);
+
+    BooleanExpression all(Function<T, BooleanExpression> map);
```

Comment:
I think probably not, because here they are actually used as a map. If you write `.any(v -> v. many().methods().here())`, everything after the `v -> v` can be moved into a preceding map stage.

Reply:
> they are actually used as a map

It does not matter how the method is implemented. Names are supposed to convey behavior/meaning, not implementation details. To me, as a user, seeing the condition which any/all array element should satisfy referred to as "map" is not helpful.

If we were creating API whose purpose is building MQL, then naming method parameters in a way that corresponds to their names in MQL would have been reasonable. But in this case we are not creating such an API, and the MQL parameter name is not `map`, it's `in`.

Reply:
Oh I see what you mean. I have changed it to `predicate`, which I updated (I think) filter to (in a forthcoming javadoc PR).